src/acl/HttpStatus.cc

```diff
@@ -132,13 +147,13 @@ aclMatchHTTPStatus(Splay<acl_httpstatus_data*> **dataptr, const Http::StatusCode
 static int
 aclHTTPStatusCompare(acl_httpstatus_data * const &a, acl_httpstatus_data * const &b)
 {
-    if (a->status1 < b->status1)
-        return 1;
+    if (a->status2 < b->status1)
+        return 1; // the entire range a is to the left of range b
```

Comment:
IMO, it would be more logical to return `-1` here (and `+1` below), but I decided not to change this (correct) old code. I did replace `a->status1` with `a->status2` above to make this code work for `a` values representing multiple status codes. For single-status ranges that this function used to handle (and still does, in addition to handling multi-status ranges), there is no difference because status1 is set to status2 for single-status ranges.